crowbar_framework/lib/crowbar/backup/base.rb

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "logger"
+
+module Crowbar
+  class Backup
+    class Base
+      attr_accessor :logger
+
+      def logger
+        @logger ||= ::Logger.new(STDOUT).tap do |logger|
+          logger.level = ::Logger::DEBUG
+        end
+      end
+
+      class << self
+        def restore_files
+          [
+            [
+              "root/.gnupg",
+              "/root/.gnupg"
+            ],
+            [
+              "root/.ssh",
+              "/root/.ssh"
+            ],
+            [
+              "configs/crowbar.json",
+              "/etc/crowbar/crowbar.json"
+            ],
+            [
+              "configs/hosts",
+              "/etc/hosts"
+            ],
+            [
+              "configs/hostname",
+              "/etc/hostname"
+            ],
+            [
+              "configs/network.json",
+              "/etc/crowbar/network.json"
```

Comment:
If we backup the /etc/crowbar/*.json files (not really sure if it's needed, but okay), then we should backup all of them, not just the one that exist by default. You can have some other files in there (at least provisioner.json).